### PR TITLE
Integration/blade volume latest

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

File: src/shaders/adam.wgsl [hunk 1]
```diff
@@ -6,13 +6,14 @@ struct Params {
...
     wd: f32,
-    _pad1: u32,
+    grad_group_size: u32,
 }
 
 var<storage, read_write> param: array<f32>;
 var<storage> grad: array<f32>;
 var<storage, read_write> m: array<f32>;
 var<storage, read_write> v: array<f32>;
+var<storage, read_write> grouped_grad_norm: array<f32>;
```
Comment (kvark) [inline src/shaders/adam.wgsl:16]: is this useful at the end of the day?

Comment (kvark) [reply src/shaders/adam.wgsl:16]: Densification needs an accumulated gradient magnitude for every point position:                                                                                                  
                                                                                                                                                                                   
  position gradient [N, 3]
          ↓
  per-point sqrt(gx² + gy² + gz²)
          ↓
  accumulate across training steps
          ↓
  split high-gradient points

  Previously blade-volume downloaded the complete [N, 3] gradient after every step and calculated this on the CPU. That forces GPU synchronization and large repeated transfers.

  The Meganeura feature instead piggybacks on Adam, which is already reading those gradients:

  - group_size = 3 means one accumulated value per XYZ point.
  - One lane calculates the three-component norm and adds it to a persistent GPU buffer.
  - Only the N accumulated values are downloaded at a densification boundary.
  - It adds no graph operation, dispatch, pipeline, or shader variant.
  - Its cost is an optional Adam buffer binding, a disabled-path conditional, and—when enabled—three extra reads plus a square root per point.

  Measured effect:

  - Bonsai gradient readback: 44.134s → 0.064s.
  - Bonsai matched segment: 84.741s → 41.164s.
  - Room readback: 25.007s → 0.007s.
  - Room growth boundary: 123.183s → 97.391s.
  - Quality remained effectively unchanged.
